integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/testsupport/util/ElasticsearchTckBackendFeatures.java

```diff
@@ -248,6 +248,28 @@ public boolean supportsExplicitRefresh() {
 
 	@Override
 	public boolean supportsVectorSearch() {
-		return false;
+		return isActualVersion(
+				esVersion -> !esVersion.isLessThan( "8.0" ),
+				osVersion -> !osVersion.isLessThan( "2.0" ),
+				osVersion -> true
+		);
+	}
+
+	@Override
+	public boolean supportsVectorSearchInsideOtherPredicates() {
+		return isActualVersion(
+				es -> false,
+				os -> true,
+				aoss -> false
```

Comment:
I find it odd the behavior of OpenSearch and Amazon OpenSearch Serverless is different here?